linux_os/guide/system/accounts/accounts-session/accounts_user_dot_no_world_writable_programs/oval/shared.xml

```diff
@@ -1,52 +1,75 @@
+    <concat>
+      <literal_component>^[^#]*</literal_component>
+      <escape_regex>
+        <variable_component var_ref="var_world_writable_programs"/>
+      </escape_regex>
+    </concat>
+  </local_variable>
+
+  <unix:file_state id="state_world_writable_programs" version="1" operator='AND'>
+    <unix:type>regular</unix:type>
+    <unix:owrite datatype="boolean">true</unix:owrite>
+  </unix:file_state>
+
+  <unix:file_object id="object_world_writable_programs" version="1">
     <unix:behaviors recurse="directories" recurse_direction="down" max_depth="-1"
-                    recurse_file_system="local"/>
-    <unix:path var_ref="var_accounts_user_dot_no_world_writable_programs_dirs"
-               var_check="at least one"/>
-    <unix:filename operation="pattern match">^\..*</unix:filename>
+      recurse_file_system="local"/>
+    <unix:path>/</unix:path>
+    <unix:filename operation="pattern match">^.*$</unix:filename>
+    <filter action="include">state_world_writable_programs</filter>
   </unix:file_object>
 
-  <!-- #### creation of state #### -->
-  <unix:file_state id="state_accounts_user_dot_no_world_writable_programs" version="1"
-                   operator='AND'>
-    <unix:gwrite datatype="boolean">false</unix:gwrite>
-    <unix:owrite datatype="boolean">false</unix:owrite>
-  </unix:file_state>
+  <ind:textfilecontent54_object id="object_accounts_user_dot_no_world_writable_programs_init_files"
+      version="2">
+    <ind:behaviors recurse="directories" recurse_direction="down" max_depth="-1"
+      recurse_file_system="local"/>
+    <ind:path var_ref="var_accounts_user_dot_no_world_writable_programs_dirs"
+      var_check="at least one"/>
+    <ind:filename operation="pattern match" var_ref="var_user_initialization_files_regex"/>
+    <ind:pattern operation="pattern match" var_ref="var_world_writable_programs_regex"
+      var_check="at least one"/>
+    <ind:instance datatype="int" operation="greater than or equal">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test id="test_accounts_user_dot_no_world_writable_programs"
+      check_existence="none_exist" check="all" version="1"
+      comment="Init files do not execute world-writable programs">
+    <ind:object object_ref="object_accounts_user_dot_no_world_writable_programs_init_files"/>
+  </ind:textfilecontent54_test>
 
-  <!-- #### creation of test #### -->
-  <unix:file_test id="test_accounts_user_dot_no_world_writable_programs" check="all"
-                  check_existence="any_exist" version="1"
-                  comment="All home directories have proper permissions">
-    <unix:object object_ref="object_accounts_user_dot_no_world_writable_programs_init_files"/>
-    <unix:state state_ref="state_accounts_user_dot_no_world_writable_programs"/>
-  </unix:file_test>
+  <external_variable id="var_user_initialization_files_regex" datatype="int" version="1" comment="User initialization files" />
```

Comment:
The CI fail looks legit:

```
 oval:ssg-var_user_initialization_files_regex:var:1 - inconsistent datatype between the variable and an associated var_ref
```

Can you take a look into it?

Reply:
Yup, it should have been `string` from the beginning. Addressed.
Thanks for the feedback!